tests/async_tests/test_interfaces.py

```diff
@@ -50,6 +50,18 @@ async def test_https_request() -> None:
         assert len(http._connections[url[:3]]) == 1  # type: ignore
 
 
+@pytest.mark.usefixtures("async_environment")
+async def test_ftp_request() -> None:
+    async with httpcore.AsyncConnectionPool() as http:
+        method = b"GET"
+        url = (b"ftp", b"example.org", 443, b"/")
+        headers = [(b"host", b"example.org")]
+        with pytest.raises(httpcore.UnsupportedProtocol):
+            http_version, status_code, reason, headers, stream = await http.request(
+                method, url, headers
+            )
```

Comment:
(Nit) I know we don't have `flake8` on this repo anymore, but mine wouldn't be happy with these unused variables (and I'd agree…), so maybe…?

```suggestion
            await http.request(method, url, headers)
```

Don't remember - what was the rationale again for removing `flake8`? I have the intuition we could bring it back with looser configuration (eg omit annoying errors such as "lines too long").

Reply:
Not sure, but yes we might be able to look at it again.